packages/collections/persistent/_vec_node.pony

```diff
@@ -1,85 +1,75 @@
-type _VecLeafNodes[A: Any #share] is Array[Array[A] val] val
-type _VecNodes[A: Any #share] is Array[_VecNode[A]] val
-
-type _VecEntries[A: Any #share] is (_VecNodes[A] | _VecLeafNodes[A])
+type _VecSubNodes[A: Any #share] is Array[_VecNode[A]]
 
 class val _VecNode[A: Any #share]
-  let _entries: _VecEntries[A]
-  let _level: U8
+  let _entries: (Array[A] val | _VecSubNodes[A] val)
 
-  new val empty(level: U8) =>
```

Comment:
Can we get some explanation here for posterity about what's going on with this part of the change?

Reply:
Since the depth is no longer stored in the node, there is no need for conversions to and from USize.